src/gui/accountsettings.cpp

```diff
@@ -511,6 +495,60 @@ void AccountSettings::slotAccountStateChanged()
     }
 }
 
+void AccountSettings::slotSpacesUpdated()
+{
+    auto spaces = accountsState()->account()->spacesManager()->spaces();
+    auto unsycnedSpaces = std::set<GraphApi::Space *>(spaces.cbegin(), spaces.cend());
+    for (const auto &f : std::as_const(FolderMan::instance()->folders())) {
+        unsycnedSpaces.erase(f->space());
+    }
+
+    // Check if we should add new spaces automagically, or only signal that there are unsynced spaces.
+    if (Theme::instance()->syncNewlyDiscoveredSpaces()) {
+        QTimer::singleShot(0, [this, unsycnedSpaces]() {
+            auto accountStatePtr = accountsState();
+
+            for (GraphApi::Space *newSpace : unsycnedSpaces) {
+                // TODO: Problem: when a space is manually removed, this will re-add it!
+                qCInfo(lcAccountSettings) << "Adding sync connection for newly discovered space" << newSpace->displayName();
+
+                const QUrl davUrl(newSpace->drive().getRoot().getWebDavUrl());
+                const QString spaceId(newSpace->drive().getRoot().getId());
+                const QString name = newSpace->displayName();
+                auto def = FolderDefinition::createNewFolderDefinition(davUrl, spaceId, name);
```

Comment:
I'd have to create a `FolderWizard::Result`, which is about as much work as creating the folder directly. I could try it though if you think it would look better.

Reply:
done